### PR TITLE
Local dev docker setup

### PR DESCRIPTION
**Pull request by zacanger**
Related: #116. This fleshes out the docker-compose file and enables one-command local development.

### CODE REVIEW

File: misc/install.sh [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
```
Comment (zacanger) [inline misc/install.sh]: This script is bourne shell compatible

File: misc/start.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+# Make sure docker-compose is installed
+if ! hash docker-compose 2>/dev/null; then
+  echo -e '\033[0;31mplease install docker\033[0m'
+  exit 1
+fi
+
+if [ -z "$(docker network ls -qf name=^entropic$)" ]; then
+  echo "Creating network"
+  docker network create entropic >/dev/null
```
Comment (zacanger) [inline misc/start.sh]: Using a network so containers can reference each other by name

File: package.json [hunk 3]
```diff
@@ -29,8 +29,10 @@
   "scripts": {
     "lint": "npm run lint-registry && npm run lint-cli",
     "lint-cli": "cd cli; npm run lint",
-    "lint-registry": "cd registry; npm run lint",
     "lint-fix": "prettier --write '**/*.js'",
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "lint-registry": "cd registry; npm run lint",
+    "postinstall": "for d in cli registry workers; do cd $d; npm i; cd -; done",
+    "start": "./misc/start.sh",
+    "test": "for d in cli registry; do cd $d; npm t; cd -; done"
```
Comment (zacanger) [inline package.json]: Changed this to run all the tests

File: docker-compose.yml [hunk 4]
```diff
@@ -8,16 +8,56 @@ services:
...
+      POSTGRES_HOST: db
+      REDIS_URL: redis://redis:6379
+      PORT: 3000
+      DEV_LATENCY_ERROR_MS: 10000
+      PGUSER: postgres
+      PGDATABASE: entropic_dev
+      CACHE_DIR: ../data/entropic-cache
+      OAUTH_GITHUB_CLIENT: gh_client_id_here
+      OAUTH_GITHUB_SECRET: gh_secret_here
+      OAUTH_PASSWORD: pw_for_encrypting_tokens_here
```
Comment (chrisdickinson) [inline docker-compose.yml]: A docker-compose newbie q: would it be possible to use `env_file` here instead? (A la [this doc](https://docs.docker.com/compose/environment-variables/#the-env_file-configuration-option))

File: misc/start.sh [hunk 5]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+set -e
+
+# Make sure docker-compose is installed
+if ! hash docker-compose 2>/dev/null; then
+  echo -e '\033[0;31mplease install docker\033[0m'
```
Comment (chrisdickinson) [inline misc/start.sh]: Now in foreboding red!

File: registry/README.md [hunk 6]
```diff
@@ -34,6 +34,10 @@ Here are the config values and what they mean:
 * `SESSION_SECRET=long_pw_for_encrypting_sessions_here`
 * `SESSION_EXPIRY_SECONDS=31536000`: how long login sessions should live
 
+## Running locally
+
+If you have Docker and Docker Compose, first `npm install`, then `npm start`.
```
Comment (chrisdickinson) [inline registry/README.md]: Could we link folks to a good place to get these prereqs?

File: docker-compose.yml (hunk 4 above)
Comment (zacanger) [reply docker-compose.yml]: Yeah absolutely, I'll change that!

File: registry/README.md [hunk 7]
```diff
@@ -34,6 +34,16 @@ Here are the config values and what they mean:
 * `SESSION_SECRET=long_pw_for_encrypting_sessions_here`
 * `SESSION_EXPIRY_SECONDS=31536000`: how long login sessions should live
 
+## Running locally
+
+* [Get Docker](https://docs.docker.com/install/)
```
Comment (zacanger) [inline registry/README.md]: Added a bit more here.

**Comment by zacanger** [comment]
Awesome, thanks for the fast review!

**Comment by zacanger** [comment]
It looks like I might've broken CI, I'll take a look at that in another PR

**Pull request merged**